### PR TITLE
NumberField -> number_field

### PR DESCRIPTION
**Pull request by fieker**
doc strings for types
removal of some very dodgy nmod_struct

### CODE REVIEW

**Comment by thofma** [comment]
Was this included in #977?

**Comment by fieker** [comment]
it wasn't...

File: src/HeckeTypes.jl [hunk 1]
```diff
@@ -371,6 +403,11 @@ mutable struct SMatSpace{T} <: Ring
   end
 end
 
+"""
+  SMat{T}
```
Comment (fingolfin) [inline src/HeckeTypes.jl:407]:
```suggestion
    SMat{T}
```

Comment (fieker) [reply src/HeckeTypes.jl:407]: when the test have passed I accept that change and merge

**Pull request merged**